common/exception/src/exception.rs

```diff
@@ -445,6 +448,12 @@ impl From<octocrab::Error> for ErrorCode {
     }
 }
 
+impl From<common_cache::DiskCacheError> for ErrorCode {
+    fn from(error: common_cache::DiskCacheError) -> Self {
+        ErrorCode::CacheInitFailed(format!("cache init failed, cause: {}", error))
```

Comment:
not sure if it is proper to convert all the `DiskCacheError` into `ErrorCode::CacheInitFailed`, 
maybe we should do the conversion explicitly where cache initialization failed?

Reply:
u r right, maybe we need do the conversion explicitly in different errors. 
And, maybe we can make conversation in `common::cache`, then `common::expection` will not depend `common::cache` ?
Do you have any suggestions ?

Reply:
Seems the mapping from `DiskCacheError` to `ErrorCode` is not "static", the conversion may depend on context. I suggest starting with explicit conversion (e.g. map_err(....))

Reply:
how about this way?
```rust
// common/cache/src/disk_cache.rs
impl From<Error> for ErrorCode {
    fn from(error: Error) -> Self {
        match error {
            Error::FileNotInCache => {
                ErrorCode::DiskCacheFileNotInCache("disk cache error: file not in cache")
            }
            Error::FileTooLarge => {
                ErrorCode::DiskCacheFileTooLarge("disk cache error: file too large")
            }
            Error::Io(err) => {
                ErrorCode::DiskCacheIOError(format!("disk cache io error, cause: {}", err))
            }
        }
    }
}

```

Reply:
neat :)  @PsiACE would you mind making common-cache depend on common-exception?

Reply:
> neat :) @PsiACE would you mind making common-cache depend on common-exception?

In fact, from my point of view, it looks strange. I have always regarded the common cache as a relatively independent module.

But it does look a bit neater this way.

Reply:
I think it is an objection :D 

@Veeupup shall we go with the " explicit conversion (e.g. ad-hoc `map_err(....)`)" way?

Reply:
using `map_err` will have some duplicate code, just like
```rust
...
        let disk_cache = Arc::new(RwLock::new(
            LruDiskCache::new(conf.disk_cache_root, conf.disk_cache_size_mb * 1024 * 1024).map_err(
                |err| match error {
                    Error::FileNotInCache => {
                        ErrorCode::DiskCacheFileNotInCache("disk cache error: file not in cache")
                    }
                    Error::FileTooLarge => {
                        ErrorCode::DiskCacheFileTooLarge("disk cache error: file too large")
                    }
                    Error::Io(err) => {
                        ErrorCode::DiskCacheIOError(format!("disk cache io error, cause: {}", err))
                    }
                },
            ),
        ));
...
        if disk_cache.contains_key(location.clone()) {
            let mut path = disk_cache.get(location).map_err(|err| match error {
                Error::FileNotInCache => {
                    ErrorCode::DiskCacheFileNotInCache("disk cache error: file not in cache")
                }
                Error::FileTooLarge => {
                    ErrorCode::DiskCacheFileTooLarge("disk cache error: file too large")
                }
                Error::Io(err) => {
                    ErrorCode::DiskCacheIOError(format!("disk cache io error, cause: {}", err))
                }
            });
            let data = read_all(&mut path)?;
...
```

can we accept the duplicate code like this?

Reply:
> I think it is an objection :D

No need to be so serious, we can add it.